datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -391,12 +391,50 @@ impl EquivalenceProperties {
     /// Updates the ordering equivalence group within assuming that the table
     /// is re-sorted according to the argument `sort_exprs`. Note that constants
     /// and equivalence classes are unchanged as they are unaffected by a re-sort.
+    /// If the given ordering is already satisfied, the function does nothing.
     pub fn with_reorder(mut self, sort_exprs: LexOrdering) -> Self {
-        // TODO: In some cases, existing ordering equivalences may still be valid add this analysis.
-        self.oeq_class = OrderingEquivalenceClass::new(vec![sort_exprs]);
+        // Filter out constant expressions as they don't affect ordering
+        let filtered_exprs = LexOrdering::new(
+            sort_exprs
+                .into_iter()
+                .filter(|expr| !self.is_expr_constant(&expr.expr))
+                .collect()
+        );
+
+        if filtered_exprs.is_empty() {
+            return self;
+        }
+
+        let mut orderings = vec![filtered_exprs.clone()];
+
+        // Preserve valid suffixes from existing orderings
+        for existing in self.oeq_class.orderings.iter() {
```

Comment:
You have the ownership of `self.oeq_class.orderings`. You can consume it while iterating maybe with mem::take, since you will build it at the end. Then you will have the ownership of `existing`, and will be able to clean clone's there:
 extended.extend(existing[filtered_exprs.len()..].iter().cloned());

Reply:
The `.cloned()` is necessary here because we can't take partial ownership of elements from the Vec inside LexOrdering - we can only get a slice reference. Rust doesn't allow splitting ownership of Vec contents, so we need to clone the elements to create new owned values.